CODE_STYLE.md

```diff
@@ -126,6 +126,25 @@ Use `re_error::format(err)` when displaying an error.
 
 
 ## C++
+For compatibility reasons we use C++11 (and nothing newer).
+We use `clang-format` to enforce most style choices (see [`.clang-format`](.clang-format)).
+
+### Initialization
+Always use `const` unless you plan on mutating it.
+
+We use `const auto x = …` for declaration because that gives symmetric code for normal constructors and static constructors:
+
+```C++
+const auto foo = SomeClass{…};
+const auto bar = SomeClass::new_xyzw(…);
+```
+
+We use `{}` for constructors (`Foo{…}` instead of `Foo(…)`).
```

Comment:
nit picky: For `constructor` or more generally for initialization? This also initializes structs in argument order 🤔. Also we can't be strict about this rule because it may choose a different overload if there's a std::initializer_list constructor present. Example I brought up on Slack yesterday:
```
auto vec_with_two_elements = std::vector<int>{3, 3};
auto vec_with_three_elements = std::vector<int>(3, 3);
```